### DISCUSSION TITLE
Format for CSV import

### DISCUSSION THREAD

**Discussion by 6illes** [netbox-community/netbox#15461] [General]
Hi, I didn't find the documentation related to CSV Import via the web interface.
Some posts are talking about REST API and Scripts but I want a CSV Import.
Thanks

**Comment by candlerb** [comment]
CSV import is documented [here](https://docs.netbox.dev/en/stable/getting-started/populating-data/#bulk-import-csvyaml).

It's also documented in the import form itself: navigate to the import page and you'll get a list of columns (mandatory ones are in bold).

![image](https://github.com/netbox-community/netbox/assets/44789/41c62266-7a09-4b0e-bd8c-1565711c50ab)

![image](https://github.com/netbox-community/netbox/assets/44789/834c67b2-84ce-45c7-818c-d7d41e29c44c)
... etc

Do beware that the CSV import format is [completely different from](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form), and incompatible with, the CSV export format.

**Comment by 6illes** [comment]
Thank you for your answer.
Now I'm facing an import problem.
Using CSV I couldn't import, so I decided to use JSON import. It's for a prefix object.
I don't succeed to import the object because it blocks at the point of the "status" parameter. It says that "Active" (but I tried also "Reserved")  is not a known status.
Thanks for help

**Reply by candlerb** [reply]
> Using CSV I couldn't import

If you show the CSV, and show the error, we can correct it. But JSON is fine too.

> it blocks at the point of the "status" parameter. It says that "Active" (but I tried also "Reserved") is not a known status.

The [status values](https://github.com/netbox-community/netbox/blob/v3.7.4/netbox/ipam/choices.py#L21-L34) are "active", "reserved" (all in lower case). These are the values stored in the database, as opposed to the values displayed to the user (which do have initial capitals).

**Reply by 6illes** [reply]
It works with "active" in lowercase
Thanks

**Reply by xkilian** [reply]
That is a common occurrence where the actual value is not apparent without looking in the database.
Especially for structures that are less common like fhrp and virtual chassis(protocols).

**Comment by 6illes** [comment]
Hi, during import (json) for a prefix, how can we declare a child ou parent relation ?
Thanks

**Reply by candlerb** [reply]
You don't.

If you insert 10.12.255.0/24 and separately insert 10.0.0.0/8 [in the same VRF], it's implicit that one is the child of the other. There's no database linkage. If you then insert 10.12.0.0/16, then implicitly it sits between the two.

**Reply by 6illes** [reply]
Thanks
